flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/diff/DiffResult.java

```diff
@@ -59,18 +63,54 @@ public String toString() {
             return "";
         }
 
-        final ToStringBuilder lhsBuilder =
-                new ToStringBuilder(left, ToStringStyle.SHORT_PREFIX_STYLE);
-        final ToStringBuilder rhsBuilder =
-                new ToStringBuilder(right, ToStringStyle.SHORT_PREFIX_STYLE);
+        final StringBuilder builder = new StringBuilder();
+        builder.append(left.getClass().getSimpleName()).append("[");
 
         diffList.forEach(
                 diff -> {
-                    lhsBuilder.append(diff.getFieldName(), diff.getLeft());
-                    rhsBuilder.append(diff.getFieldName(), diff.getRight());
+                    try {
+                        JsonNode before =
+                                objectMapper.readTree(
+                                        objectMapper.writeValueAsString(diff.getLeft()));
+                        JsonNode after =
+                                objectMapper.readTree(
+                                        objectMapper.writeValueAsString(diff.getRight()));
+                        JsonNode jsonDiff = JsonDiff.asJson(before, after);
+                        jsonDiff.forEach(
+                                row -> {
+                                    if (row.get("path").asText().equals("/")) {
+                                        builder.append(diff.getFieldName())
+                                                .append(" : ")
+                                                .append(before)
+                                                .append(" -> ")
+                                                .append(after)
+                                                .append(", ");
+                                    } else {
+                                        builder.append(diff.getFieldName())
+                                                .append(
+                                                        row.get("path")
+                                                                .asText()
+                                                                .replaceAll("/", "."))
+                                                .append(" : ")
+                                                .append(before.at(row.get("path").asText()))
+                                                .append(" -> ")
+                                                .append(after.at(row.get("path").asText()))
+                                                .append(", ");
+                                    }
+                                });
+                        builder.setLength(builder.length() - 2);
+                    } catch (JsonProcessingException je) {
```

Comment:
Maybe we can be a bit more careful here and catch all kinds of exceptions, to be on the safe side :)